src/m365/spo/commands/customaction/customaction-get.ts

```diff
@@ -136,70 +151,88 @@ class SpoCustomActionGetCommand extends SpoCommand {
     }
+    else {
+      const filter: string = options.id ?
+        `('${formatting.encodeQueryParameter(options.id as string)}')` :
+        `?$filter=Title eq '${formatting.encodeQueryParameter(options.title as string)}'`;
+
+      const requestOptions: any = {
+        url: `${options.webUrl}/_api/${options.scope}/UserCustomActions${filter}`,
+        headers: {
+          accept: 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+
+      if (options.id) {
+        const result = await request.get<CustomAction>(requestOptions);
+        return result;
+      }
 
-    return request
-      .get<{ value: CustomAction[] }>(requestOptions)
-      .then((res: { value: CustomAction[] }): Promise<CustomAction> => {
-        if (res.value.length === 1) {
-          return Promise.resolve(res.value[0]);
-        }
+      const result = await request.get<{ value: CustomAction[] }>(requestOptions);
 
-        if (res.value.length === 0) {
-          return Promise.reject(`No user custom action with title '${options.title}' found`);
-        }
+      if (result.value.length === 0) {
+        throw `No user custom action with title '${options.title}' found`;
+      }
 
-        return Promise.reject(`Multiple user custom actions with title '${options.title}' found. Please disambiguate using IDs: ${res.value.map(a => a.Id).join(', ')}`);
-      });
+      if (result.value.length > 1) {
+        throw `Multiple user custom actions with title '${options.title}' found. Please disambiguate using IDs: ${result.value.map(a => a.Id).join(', ')}`;
+      }
+
+      return result.value[0];
+    }
   }
 
   /**
    * Get request with `web` scope is send first. 
    * If custom action not found then 
    * another get request is send with `site` scope.
    */
-  private searchAllScopes(options: Options): Promise<CustomAction> {
-    return new Promise<CustomAction>((resolve: (customAction: CustomAction) => void, reject: (error: any) => void): void => {
+  private async searchAllScopes(options: Options): Promise<CustomAction> {
+    try {
       options.scope = "Web";
 
-      this
-        .getCustomAction(options)
-        .then((webResult: CustomAction): void => {
-          if (webResult["odata.null"] !== true) {
-            return resolve(webResult);
-          }
+      const webResult = await this.getCustomAction(options);
+      if (webResult["odata.null"] !== true) {
+        return webResult;
+      }
 
-          options.scope = "Site";
-          this
-            .getCustomAction(options)
-            .then((siteResult: CustomAction): void => {
-              return resolve(siteResult);
-            }, (err: any): void => {
-              reject(err);
-            });
-        }, (err: any): void => {
-          reject(err);
-        });
-    });
+      options.scope = "Site";
+      const siteResult = await this.getCustomAction(options);
+      return siteResult;
+
+    }
+    catch (err) {
+      throw err;
```

Comment:
If we're rethrowing the error, what's the purpose of this try..catch statement? Let's remove it.